findsecbugs-samples-java/src/test/java/testcode/sqli/VertxSqlClient.java

```diff
@@ -0,0 +1,33 @@
+package testcode.sqli;
+
+import io.vertx.sqlclient.SqlClient;
+import io.vertx.sqlclient.SqlConnection;
+
+public class VertxSqlClient {
+
+    public void injection1(SqlClient client, String injection) {
+        client.query(injection);
```

Comment:
Example are minimalist but I think the vulnerability type for these API is well known.

Reply:
Yes the vert.x API is fluent so there's just 1 way to pass a sql statement to be executed, there are 2 classes where this happens:

* SqlClient - means a connection managed client where users run queries ad-hoc without worring to get a connection and returning it to the pool.
* SqlConnection - means that the user is being responsible to control the connection and return it to the pool.